source/geod24/LocalRest.d

```diff
@@ -303,22 +303,54 @@ public void sleep (Duration timeout)
         dg = If non-null, this delegate will be called when the timer fires
         periodic = Speficies if the timer fires repeatedly or only once
 
+    Returns:
+        a Timer instance with the ability to stop the timer
+
 *******************************************************************************/
 
-public void setTimer (Duration timeout, void delegate() dg,
+public Timer setTimer (Duration timeout, void delegate() dg,
     bool periodic = false)
 {
     assert(scheduler !is null, "Cannot call this delegate from the main thread");
     assert(dg !is null, "Cannot call this delegate if null");
-    scheduler.schedule(
-    ()
+
+    Timer timer = Timer(timeout, dg, periodic);
+    scheduler.schedule(() { timer.run(); });
+    return timer;
+}
+
+/// Simple timer
+public struct Timer
+{
+    private Duration timeout;
+    private void delegate () dg;
+    // Using a pointer so Timer may be copied and stop() will still work
+    private bool* periodic;
```

Comment:
Since it has an identity, I'm wondering why we don't use classes for this ?

Reply:
I'd better change it to `class`
because it should have each pointer periodic.

Reply:
What he means is this:

```D
final class Timer
{
    bool periodic;

    ....
}
```

With this `periodic` does not need to be a pointer.

Reply:
> With this periodic does not need to be a pointer.

Oh, thank you. the right answer 'final class'

I was testing in `agora`, and I experienced a change of reference address...

Reply:
It shouldn't really matter if it's final or not, it would work either way. But with `final` you cannot inherit from this class.

By the way, what code failed for you?